UserDocs.md

```diff
@@ -14,245 +14,233 @@ You may wish create a Quilt-specific [profile](https://docs.aws.amazon.com/cli/l
 
-You can annotate your object with custom metadata by passing a `dict`
-to `meta=`. Object metadata are indexed and searchable by the helium API.
+If you would like to save or read a local package, you can use the following methods to do so.
 
-For more on metadata, see [Metadata and search](#metadata-and-search).
+	# save an empty package to the local registry
+	p = t4.Package()
+	tophash = p.build("example-package")
 
-#### `helium.get_file(src, dest [, snapshot=None | version=None])`
-Retrieves `src` object from T4, and writes to the `dest` file on your local disk.
+	# load the package from a local registry
+	p = t4.Package("example-package", hash=tophash)
 
-`snapshot=` and `version=` optional parameters, described in `helium.get()`.
+Packages are versioned by top hashes. Every time you build a package you are returned a top hash. To load a specific version of a package from a registry, as here, you can pass the top hash of the desired package version to the `hash` parameter. If you do not specify a top hash, the latest version will be retrieved.
 
-#### `helium.put_file(src, dest, meta={})`
-Writes the local file `src` to `dest` in S3.
 
-Writes the in-memory object `src` to the path `dest` in S3.
-Transparently serializes some object types.
-See [Serialization](#serialization) for details.
+### Publishing a package to T4
+Once you have a package manifest you are happy with, you are ready to publish. To publish the package to T4, run the following command:
 
-You can annotate your object with custom metadata by passing a `dict`
-to `meta=`. Object metadata are indexed and searchable by the helium API.
+    p.push("example-package", "s3://name-of-your-t4-bucket/my/package/path")
 
-As with `helium.put`, you may specify metadata for the object by pass a dictionary of key-value pairs to the `meta` parameter. For more on metadata, see [Metadata and search](#metadata-and-search).
 
-#### `helium.delete(path)`
-Deletes the object at `path` (does not work on folders).
+This will:
+* Copy the manifest to T4. The manifest will be stored in your T4 bucket’s package registry by both its top hash and the name you provide (`"example-package"`).
+* Copy the package contents to `/my/package/path` in your bucket according to their logical keys.
 
-### Snapshots and history
+Once the dataset has been materialized on T4, it will be available to anyone else with access to that T4 bucket.
 
-#### `helium.snapshot(path, message)`
-Creates a snapshot of the T4 object at `path` with commit message `message`.
 
-#### `helium.list_snapshots(bucket, contains=None)`
-List all snapshots in `bucket`. `contains=` is an optional parameter that limits
-the results to snapshots that contain the specified prefix.
+### Pulling a package from T4
+To pull a package that has been published to T4:
 
-Returns path, hash, timestamp, and snapshot message.
+	# load the package from the registry
+	p = t4.Package("nice-name", registry="name-of-your-t4-bucket")
```

Comment:
I would put `s3://` in the registry name so people are 100% sure that's required